### PR TITLE
Feature: Add Recurring add-on product overlay to Reports widget 

### PR DESCRIPTION
**Pull request by JoshuaHungDinh**
# Description

This PR implements a modal-like overlay on the Reports Widget in the admin dashboard. Its purpose is to provide users with information about the recurring add-on if they are unfamiliar with it. The dialog box also includes a button that opens a new window and directs users to a page where they can read about and purchase the Recurring add on. The overlay and dialog will only appear to users who have not received any donations yet.

## Affects

Wordpress Admin page Reports Widget

## Visuals

https://github.com/impress-org/givewp/assets/75056371/20efce11-4441-43b3-9967-f51b8bebba04

## Testing Instructions

- Ensure you have no donations.
- Visit the Admin dashboard and view the Reports Widget. 
- Verify the dialog box is displayed and verify the UTM link is correct and opens Recurring add-on page in a new window.
- Make a donation.
- Revisit the widget and verify the dialog is no longer displayed.

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [x] Reviewed by the designer (if follows a design)
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204875481837188

### CODE REVIEW

File: src/Views/Admin/DashboardWidgets/Reports.php [hunk 1]
```diff
@@ -73,15 +68,11 @@ public function enqueue_scripts($base)
                 'allTimeStart' => $this->get_all_time_start(),
                 'currency' => give_get_currency(),
                 'testMode' => give_is_test_mode(),
+                'assets' => GIVE_PLUGIN_URL,
```
Comment (glaubersilva) [inline src/Views/Admin/DashboardWidgets/Reports.php:71]: @JoshuaHungDinh I notice that in the JS files, you are using it like that: `${assets}assets/dist/images/admin/white-external-icon.svg`

So, we can let it more clear replace this line with something like this: 

`'assetsUrl' => GIVE_PLUGIN_URL . 'assets/',`

And then we can use that in the front like that:

`${assetsUrl}dist/images/admin/white-external-icon.svg`

**Review by glaubersilva — CHANGES_REQUESTED** [review]

File: src/Views/Admin/DashboardWidgets/Reports.php (hunk 1 above)
Comment (JoshuaHungDinh) [reply src/Views/Admin/DashboardWidgets/Reports.php]: Good thought! Thanks @glaubersilva Resolved: 7e8db152cdff4f6db93c7a83df6bca3e5113e6ea

**Review by glaubersilva — APPROVED** [review]
@JoshuaHungDinh Thanks for the update!

**Review by rickalday — APPROVED** [review]
Passed manual QA tests

**Pull request merged**